.github/workflows/CI.yml

```diff
@@ -215,6 +215,271 @@ jobs:
+          key: gvsbuild-${{ env.gvsbuildupdate }}-${{ env.gvsbuildref }}
+
+      - name: (GTK binaries) checkout gvsbuild
+        if: steps.cache.outputs.cache-hit != 'true'
+        uses: actions/checkout@v2
+        with:
+          repository: wingtk/gvsbuild
+          ref: ${{ env.gvsbuildref }}
+          path: gvsbuild
+
+      # Remove the preinstalled git, it causes errors related to cygwin.
+      # Note this needs to happen after we checked out the projects.
+      - name: (GTK binaries) remove git binary
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: rmdir "C:\Program Files\Git\usr\bin" /s /q
+        shell: cmd
+
+      - name: (GTK binaries) run gvsbuild
+        if: steps.cache.outputs.cache-hit != 'true'
+        working-directory: gvsbuild
+        run: python .\build.py build -p=x64 --vs-ver=16 --msys-dir=C:\msys64 gtk3
+
+      - name: Setup env
+        run: |
+          echo "PKG_CONFIG=C:\gtk-build\gtk\x64\release\bin\pkgconf.exe" >> $GITHUB_ENV
+          echo "C:\gtk-build\gtk\x64\release\bin" >> $GITHUB_PATH
+        shell: bash
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          default: true
+
+      # ATK
+
+      - name: atk test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --manifest-path atk/Cargo.toml --features "v2_34"
+
+      # FAILS: thread 'cross_validate_constants_with_c' panicked at 'configured compiler: Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." }', atk\sys\tests\abi.rs:131:30
+      #- name: atk-sys test
+      #  uses: actions-rs/cargo@v1
+      #  with:
+      #    command: test
+      #    args: --manifest-path atk/sys/Cargo.toml --features "v2_34"
+
+      - name: atk build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --manifest-path atk/Cargo.toml --features "v2_34"
+
+      # Cairo
+
+      # HANGS: test context::tests::clip_rectange ... test context::tests::clip_rectange has been running for over 60 seconds
+      #- name: cairo test
+      #  uses: actions-rs/cargo@v1
+      #  with:
+      #    command: test
+      #    args: --manifest-path cairo/Cargo.toml --features "png,pdf,svg,ps,use_glib,v1_16,freetype,script,win32-surface"
+
+      - name: cairo-sys test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --manifest-path cairo/sys/Cargo.toml --features "png,pdf,svg,ps,use_glib,v1_16,freetype,script,win32-surface"
+
+      - name: cairo build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --manifest-path cairo/Cargo.toml --features "png,pdf,svg,ps,use_glib,v1_16,freetype,script,win32-surface"
+
+      # GDK
+
+      - name: gdk test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --manifest-path gdk/Cargo.toml --features v3_24
+
+      # FAILS: abi check
```

Comment:
Most of the abi tests are failing,  see #54. I would try to keep them in-sync if possible

Reply:
And if different sys crates fail on Windows than on Linux that should be documented and investigated shortly at least.